ci/test/01_base_install.sh

```diff
@@ -84,6 +84,29 @@ if [[ ${USE_MEMORY_SANITIZER} == "true" ]]; then
   rm -rf /msan/llvm-project
 fi
 
+if [[ ${BARE_METAL_RISCV} == "true" ]]; then
+    ${CI_RETRY_EXE} git clone --depth=1 https://github.com/riscv-collab/riscv-gnu-toolchain -b 2024.11.22 /riscv/gcc
+    cd /riscv/gcc
+    ./configure --prefix=/opt/riscv-ilp32 --with-arch=rv32gc --with-abi=ilp32
+    make -j "$MAKEJOBS"
+    make install
+
+    ${CI_RETRY_EXE} git clone --depth=1 https://sourceware.org/git/newlib-cygwin.git -b topic/3.6 /riscv/newlib
+    cd /riscv/newlib
+    mkdir build && cd build
+    ../configure \
+        --target=riscv32-unknown-elf --disable-newlib-io-float --enable-newlib-io-long-long --enable-newlib-io-long-double --with-arch=rv32gc --with-abi=ilp32 --disable-shared --disable-multilib\
```

Comment:
Are you sure enabling i/o for long-double is needed? i don't believe we use this type anywhere.